src/Compilers/Core/Portable/StrongName/CryptoBlobParser.cs

```diff
@@ -211,9 +213,15 @@ public unsafe static bool TryGetPublicKey(ImmutableArray<byte> blob, out Immutab
                 // Info (corresponds to RSAPUBKEY struct in wincrypt.h)
                 var magic = blobReader.ReadUInt32();
                 var bitLen = blobReader.ReadUInt32();
+
+                // We need the exponent as both a byte array and
+                // as an integer depending on if we're doing public sign or
+                // strong name signing with private keys.
+                var expBytes = blobReader.ReadBytes(4);
+                blobReader.Offset -= 4;
```

Comment:
Careful about endianity though. BitConverter.ToUInt32 is little-endian, i think.